python/ray/dag/compiled_dag_node.py

```diff
@@ -1075,6 +1192,202 @@ def _get_or_compile(
         self._dag_submitter.start()
         self._dag_output_fetcher.start()
 
+    def _generate_dag_operation_graph_node(
+        self,
+    ) -> Dict["ray.actor.ActorHandle", List[List[DAGOperationGraphNode]]]:
+        """
+        Generate READ, COMPUTE, and WRITE operations for each DAG node.
+
+        Returns:
+            A dictionary that maps an actor handle to a list of lists of
+            DAGOperationGraphNode. For the same actor, the index of the
+            outer list corresponds to the index of the ExecutableTask in
+            the list of `executable_tasks` in `actor_to_executable_tasks`.
+            In the inner list, the order of operations is READ, COMPUTE,
+            and WRITE.
```

Comment:
nit: maybe add some diagram for ease of understanding

```
[
    [READ COMPUTE WRITE] # local_idx 0
    [READ COMPUTE WRITE] # local_idx 1
]
```

Reply:
Added https://github.com/ray-project/ray/pull/46911/commits/45fb99fa9c2ebe3c80ad587888c1cb985ef9cc1c